src/naarad/__init__.py

```diff
@@ -426,10 +427,16 @@ def _process_naarad_config(self, config, analysis):
             naarad.utils.parse_basic_metric_options(config, section)
           sar_metrics = naarad.utils.get_all_sar_objects(metrics, infile, hostname, output_directory, label, ts_start,
                                                          ts_end, None)
+          if ts_start is not None and sar_metric.ts_start is None and ts_end is not None and sar_metric.ts_end is None:
```

Comment:
@feng-tao , might be better to separate out the ts_start and ts_end conditions. Users should be able to specify only ts_start or ts_end and still have that take effect.